src/send/SendEnterAmount.tsx

```diff
@@ -100,18 +105,31 @@ function SendEnterAmount({ route }: Props) {
   const { t } = useTranslation()
   const { defaultTokenIdOverride, origin, isFromScan, forceTokenId } = route.params
   const supportedNetworkIds = getSupportedNetworkIdsForSend()
-  const tokens = useSelector((state) =>
+  const supportedSendTokens = useSelector((state) =>
     tokensWithNonZeroBalanceAndShowZeroBalanceSelector(state, supportedNetworkIds)
   )
+  const supportedJumpstartTokens = useSelector(jumpstartSendTokensSelector)
   const lastUsedTokenId = useSelector(lastUsedTokenIdSelector)
 
+  const tokens = useMemo(() => {
+    return origin === SendOrigin.Jumpstart ? supportedJumpstartTokens : supportedSendTokens
+  }, [supportedSendTokens, supportedJumpstartTokens, origin])
   const defaultToken = useMemo(() => {
     const defaultToken = tokens.find((token) => token.tokenId === defaultTokenIdOverride)
     const lastUsedToken = tokens.find((token) => token.tokenId === lastUsedTokenId)
 
     return defaultToken ?? lastUsedToken ?? tokens[0]
   }, [tokens, defaultTokenIdOverride, lastUsedTokenId])
 
+  const jumpstartLink = useMemo(() => {
```

Comment:
This got me thinking. We need to create a new private key for each invitation (all the rewards sent with the same private key can be claimed with one link, so we shouldn't create more than one invitation per key).
I'm not sure if with this code it could be reused for different invitations. I think we are fine, because we are creating a new `SendEnterAmount` component in every attempt, right? But just double checking I'm not missing some case.

I would put inside the `onPressReview` the code to get a new private key in each attempt and avoid confusions.

Reply:
we need the public key to prepare the transactions, which is done before executing `onPressReview` so we cannot generate the private key in that handler. it's safe to generate the key pair on mount of this screen, since the user will leave this screen once the transaction is completed

Reply:
Oh, you are right, we need to prepare the transaction.
I think this is fine, since we don't allow the user to go back to this screen after they execute the deposit.